recipes/gmp/all/conanfile.py

```diff
@@ -69,19 +69,19 @@ def package(self):
         with tools.chdir(self._source_subfolder):
             autotools = self._configure_autotools()
             autotools.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
-        # remove la files
-        for la_name in ['libgmp.la', 'libgmpxx.la']:
-            la = os.path.join(self.package_folder, "lib", la_name)
-            if os.path.isfile(la):
-                os.unlink(la)
+        os.unlink(os.path.join(self.package_folder, "lib", "libgmp.la"))
+        if self.options.enable_cxx:
+            os.unlink(os.path.join(self.package_folder, "lib", "libgmpxx.la"))
 
     def package_id(self):
         del self.info.options.run_checks  # run_checks doesn't affect package's ID
 
     def package_info(self):
+        self.cpp_info.components["libgmp"].libs = ["gmp"]
+        self.cpp_info.components["libgmp"].names["pkg_config"] = ["gmp"]
```

Comment:
@danimtb how does this work? `names` for `pkg_config` at a component level

Reply:
yes, the syntax is allowed. However, the `pkg_config` generator still does not handle components. Do you know what are the equivalent to components in pkg-config @madebr?

Reply:
pkg-config pc files and targets have a 1:1 mapping. Each file represents one target (and its dependencies).
See https://people.freedesktop.org/~dbn/pkg-config-guide.html for an example.

In the case of gmp, 2 pc files: `gmp.pc` and `gmpxx.pc` should be created.